### PR TITLE
MockOracle refactor

### PR DESCRIPTION
**Pull request by Kukovec**
- [x] Tests added for any new code
- [x] Ran `make fmt-fix` (or had formatting run automatically on all files edited)
- [ ] Documentation added for any new functionality
- [ ] [Entries added to `./unreleased/`][changelog format] for any new functionality

Following the pattern laid out in #2709 for `IntOracle`, this PR introduces the same alterations to `MockOracle`.

[changelog format]: https://github.com/informalsystems/apalache/blob/main/CONTRIBUTING.md#how-to-record-a-change

### CODE REVIEW

File: tla-bmcmt/src/test/scala/at/forsyte/apalache/tla/bmcmt/stratifiedRules/aux/oracles/TestIntOracle.scala [hunk 1]
```diff
@@ -126,7 +126,8 @@ class TestIntOracle extends AnyFunSuite with BeforeAndAfterEach with Checkers {
   }
 
   // We cannot test getIndexOfChosenValueFromModel without running the solver
-  test("getIndexOfChosenValueFromModel recovers the index correctly") {
+  // Ignored until we figure out why it's killing GH CLI
+  ignore("getIndexOfChosenValueFromModel recovers the index correctly") {
```
Comment (thpani) [inline tla-bmcmt/src/test/scala/at/forsyte/apalache/tla/bmcmt/stratifiedRules/aux/oracles/TestIntOracle.scala:130]: Can we
- [ ] refer to the relevant issue here, and
- [ ] leave a note under that issue to (re-)enable these tests?

File: tla-bmcmt/src/test/scala/at/forsyte/apalache/tla/bmcmt/stratifiedRules/aux/oracles/TestMockOracle.scala [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+      forAll(gen) { case (fixed, assertionsIfTrue, assertionsIfFalseOpt) =>
+        val oracle = MockOracle.create(fixed)
+        val caseEx: TlaEx = oracle.caseAssertions(initScope, assertionsIfTrue, assertionsIfFalseOpt)
+        caseEx == assertionsIfTrue(fixed).build
+      }
+
+    check(prop, minSuccessful(1000), sizeRange(4))
+  }
+
+  // We don't actually need the solver in MockOracle
```
Comment (thpani) [inline tla-bmcmt/src/test/scala/at/forsyte/apalache/tla/bmcmt/stratifiedRules/aux/oracles/TestMockOracle.scala:104]: By this you mean, we are passing it, but it's not actually invoked?

File: tla-bmcmt/src/test/scala/at/forsyte/apalache/tla/bmcmt/stratifiedRules/aux/oracles/TestSparseOracle.scala [hunk 3]
```diff
@@ -70,7 +70,8 @@ class TestSparseOracle extends AnyFunSuite with BeforeAndAfterEach with Checkers
   // which should have its own tests
 
   // We cannot test getIndexOfChosenValueFromModel without running the solver
-  test("getIndexOfChosenValueFromModel recovers the index correctly") {
+  // Ignored until we figure out why it's killing GH CLI
+  ignore("getIndexOfChosenValueFromModel recovers the index correctly") {
```
Comment (thpani) [inline tla-bmcmt/src/test/scala/at/forsyte/apalache/tla/bmcmt/stratifiedRules/aux/oracles/TestSparseOracle.scala:74]: Can we
- [ ] refer to the relevant issue here, and
- [ ] leave a note under that issue to (re-)enable these tests?

**Review by thpani — APPROVED** [review]
LGTM

We should track re-enabling these commented-out tests though, or we might forget.

**Comment by shonfeder** [comment]
Looks like the CI failures were just transient issues in the GH runners. I'll fix the comments and get this merged.

**Pull request merged**